docs/devs/Writing-New-Module.md

```diff
@@ -172,4 +269,46 @@ And for asynchronous callbacks, after `libtuv` calls your native function, if yo
 
 For asynchronous callbacks, you must consider the lifetime of JS-side callback objects. The lifetime of JS-side callback object should be extended until the native-side callback is really called. You can use `iotjs_reqwrap_t` and `iotjs_handlewrap_t` to achieve this.
 
-(Work In Progress)
+## Writing "Mixed" Modules
+
+Modules could be a combination of JS and native code. In that case the Javascript file must
+export the objects of the module. In such cases the native part will be hidden.
+
+For simple explanation, `console` module will be used as an example.
+
+```
+src
+ |-- js
+      |-- console.js
+ |-- modules
+      |-- iotjs_module_console.c
+      |-- modules.json
+```
+
+modules.json
+```json
+{
+  "modules": {
+    ...
+    "console": {
+      "native_files": ["modules/iotjs_module_console.c"],
+      "init": "InitConsole",
+      "js_file": "js/console.js",
+      "require": ["util"]
+    },
+    ...
+  }
+}
+```
+
+### Using native module in JavaScript module
+
+Logging to console needs native functionality, so `console` JavaScript module in `src/js/console.js` passes its arguments into native handler like:
+
+```javascript
+Console.prototype.log = native.stdout(util.format.apply(this, arguments) + '\n');
+```
+
+Where `native` is the JS object returned by the native `InitConsole` function in `iotjs_module_console.c`.
```

Comment:
Is it better to specify a native variable name when using a native module in the JS module? I am confused. For example, in `gpio.js`, the native binding variable is `gpio`(not native or native.gpio), but `dns.js` is `native`. 
And how do I use more than two native module in JS module?

Reply:
We changed the function wrapper to use an extra `native` parameter. If `native` is not undefined than the module has parts written in C. The `native` variable contains the return value of the `init` function.

> And how do I use more than two native module in JS module?

You can use only the `require()` and use what a module exports. This is intentional. Our idea was to make it possible to hide implementation details from module users. If you write a module with both js and native files, then it is the javascript file's responsibility to export the public objects. The native files of that module is only available for that module. For such use cases where a native part must be shared, you can create a module that only contains native files. If there is no JS file presented then the return value of the init function will be exported.

```
{
  "modules": {
    "common_native_module": {
      "native_files": ["common.c"],
      "init": "InitCommonThings"
    }
  }
}
```

my_module1.js:
```js
var common = require('common_native_module');
if (native) {
 // private native part
}
...
```

my_module2.js:
```js
var common = require('common_native_module');
if (native) {
 // private native part
}
...
```